doc/source/dev/contributor/devpy_test.rst

```diff
@@ -0,0 +1,145 @@
+:orphan:
+
+.. _devpy-runtest:
+
+===========================
+Running SciPy Tests Locally
+===========================
+
+Basic test writing and execution from within the Python interpreter is
+documented in the `NumPy/SciPy Testing Guidelines`_. This page includes
+information about running tests from the command line using SciPy's
+``dev.py`` command line tool. *Note: Before beginning, ensure that* |pytest|_
+*is installed.*
+
+To run all tests, navigate to the root SciPy directory at the command
+line and execute
+
+::
+
+   python dev.py test -v
+
+where ``-v`` activates the ``--verbose`` option. This builds SciPy (or
+updates an existing build) and runs the tests.
+
+To run tests on a particular submodule, such as ``optimize``, use the
+``--submodule`` option:
+
+::
+
+   python dev.py test -v -s optimize
+
+To run a particular test module, use the ``--test`` option:
+
+::
+
+   python dev.py test -v -t scipy.<module>.tests.<test_file>
+
+Example for |test-linprog|_ file tests, run:
+
+::
+
+   python dev.py test -v -t scipy.optimize.tests.test_linprog
+
+To run a test class:
+
+::
+
+   python dev.py test -v -t scipy.<module>.tests.<test_file>::<TestClass>
+
+Example for ``TestLinprogRSCommon`` class from ``test_linprog.py``:
+
+::
+
+   python dev.py test -v -t scipy.optimize.tests.test_linprog::TestLinprogRSCommon
+
+To run a particular test:
+
+::
+
+   python dev.py test -v -t scipy.<module>.tests.<test_file>::<test_name>
+
+Example for ``test_unknown_solvers_and_options`` from ``test_linprog.py``:
+
+::
+
+   python dev.py test -v -t scipy.optimize.tests.test_linprog::test_unknown_solvers_and_options
+
+For tests within a class, you need to specify the class name and the test name:
+
+::
+
+   python dev.py test -v -t scipy.<module>.tests.<test_file>::<TestClass>::<test_name>
+
+Example:
+
+::
+
+   python dev.py test -v -t scipy.optimize.tests.test_linprog::TestLinprogRSCommon::test_nontrivial_problem_with_guess
+
+
+Other useful options include:
```

Comment:
There is no `doc,bench,pep8,mypy` for `python dev.py test`. And I would not talk about `refguide-check` here.